src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,144 @@
+(ns puppetlabs.services.legacy-routes.legacy-routes-core
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [ring.util.codec :as ring-codec]
+            [compojure.route :as route]
+            [clojure.tools.logging :as log]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; v3 => v4 header translation support functions
+
+; TODO: Improve the naive parsing here with a real library like
+; https://github.com/ToBeReplaced/http-accept-headers
+(defn- map-accept-raw-to-binary
+  "Convert Accept: raw media types to Accept: binary.  NOTE: This method does
+  not conform to RFC-2616 Accept: header specification, but then again
+  neither does Accept: raw or Accept: binary
+
+  The split on `,` is naive, the method should take into account `;` parameters
+  and the such to be RFC compliant, but puppet agent does not use this format
+  so implementing the full spec is unnecessary."
+  [request]
+  (if-let [accept (get-in request [:headers "accept"])]
+    (let [vals (str/split accept #"\s*,\s*")]               ; BAD! See RFC-2616
+      (assoc-in request [:headers "accept"]
+        (str/join ", " (distinct (replace {"raw" "binary"} vals)))))
+    request))
+
+(defn- map-accept-s-pson-to-binary
+  "Convert Accept: s, pson media types to Accept: binary given a ring request
+  map.  This method should only be used to munge file_bucket_file API requests
+  from legacy agents speaking to a contemporary master."
+  [request]
+  (if-let [accept (get-in request [:headers "accept"])]
+    (let [v4-accept-val "binary"
+          vals (str/split accept #"\s*,\s*")                ; BAD! see RFC-2616
+          munged-accept (->> vals
+                          (replace {"s" v4-accept-val "pson" v4-accept-val})
```

Comment:
@jeffmccune Ok, thanks.  So my preference would still be - unless there's some dangerous reason not to do so - to just collapse the `map-accept-raw-to-binary` and `map-accept-s-pson-binary` functions into a single function which walks over the accept header values in the request once, substituting any instances of `s` or `raw` with `binary` and preserving everything else.  That function could just be called by `request-compatibility-wrapper` so that it all routes are automatically covered - to avoid us having to worry about which ones might use `s` vs `raw` and in what scenarios, etc.  Also, we'd only have to look at one spot in the code to know where any `accept` header translation is being done.

Would be curious to hear what others' thoughts are on this.  @dankreek - what do you think?